lib/nebulex/adapters/local/generation.ex

```diff
@@ -400,6 +499,34 @@ defmodule Nebulex.Adapters.Local.Generation do
     {false, state}
   end
 
+  defp maybe_generation_cleanup_impl(
+         %__MODULE__{
+           meta_tab: meta_tab,
+           backend: backend,
+           generation_cleanup_ref: generation_cleanup_ref,
+           generation_max_size: generation_max_size,
+           generation_created_at: generation_created_at,
+           generation_start_timeout: generation_start_timeout,
```

Comment:
Please check my comment above, there are two timers, and `:generation_start_timeout` should run separately, it shouldn't be validated here, but in a different `handle_info` clause when the `:generation_start_timeout` occurs.